src/app/ApplicationTemplate.Shared/Framework/DataLoader/OnDataLostDataLoaderStrategy.cs

```diff
@@ -25,7 +25,7 @@ public override async Task<object> Load(CancellationToken ct, IDataLoaderRequest
 		{
 			var result = await base.Load(ct, request);
 
-			if (_data != null)
+			if (_data != null && !ReferenceEquals(_data, result))
```

Comment:
The change makes sense, but add comments justifying the ReferenceEquals check

Reply:
I was wondering the same but ReferenceEquals is quite obvious, isn't it ?